storagenode/orders/sender.go

```diff
@@ -120,15 +143,24 @@ func (sender *Sender) runOnce(ctx context.Context) (err error) {
 		ctx, cancel := context.WithTimeout(ctx, sender.config.Timeout)
 		defer cancel()
 
+		ch := make(chan archiveRequest, batchSize)
+		done := make(chan struct{})
```

Comment:
Let's use errgroup here to propagate the error and make it clearer the goroutine is managed and waited for.